modin/engines/base/frame/partition.py

```diff
@@ -146,7 +160,45 @@ def mask(self, row_indices, col_indices):
         PandasFramePartition
             New `PandasFramePartition` object.
         """
-        pass
+        if (
+            (isinstance(row_indices, slice) and row_indices == slice(None))
+            or (
+                not isinstance(row_indices, slice)
+                and isinstance(self._length_cache, int)
+                and len(row_indices) == self._length_cache
+            )
+        ) and (
+            (isinstance(col_indices, slice) and col_indices == slice(None))
+            or (
+                not isinstance(col_indices, slice)
+                and isinstance(self._width_cache, int)
+                and len(col_indices) == self._width_cache
+            )
+        ):
+            return self.__copy__()
+
+        new_obj = self.add_to_apply_calls(
+            lambda df: pandas.DataFrame(df.iloc[row_indices, col_indices])
```

Comment:
Since the partition class called `PandasFramePartition` we assume that the underlying partition data is pandas DataFrame

Reply:
> why we need to cast to pandas.DataFrame here anyway?

`.iloc` may produce scalar in case of single-value result, however, partitions does not support anything more than the pandas DataFrame of Series:
https://github.com/modin-project/modin/blob/653d5b235ffe373b1416fc8049c1080db89bc156/modin/engines/ray/pandas_on_ray/frame/partition.py#L177-L179

Reply:
> each Frame should have an underlying partition class

Yes, it has, but it points to the actually partition class, not to its payload type:
https://github.com/modin-project/modin/blob/653d5b235ffe373b1416fc8049c1080db89bc156/modin/engines/ray/pandas_on_ray/frame/partition_manager.py#L139

Reply:
In that case I suggest checking if it's scalar and only then wrap in the `DataFrame` (less pessimization, more performance).
Also ideally I would like to get `pandas.DataFrame` as a class attribute from somewhere (i.e. not hardcode so it keeps working regardless of backend).

Reply:
>`.iloc` may produce scalar in case of single-value result

found out that it does so only if masked indices are scalars itself, so replaced explicit result conversion to pandas DataFrame by explicit indices conversion to list-like values
```python
>>> df.iloc[0, 0]
1
>>> df.iloc[[0], [0]]
   a
0  1
```